src/main/java/com/lambdaworks/redis/protocol/CommandHandler.java

```diff
@@ -213,8 +213,23 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buffer) throws Interrup
 
             WithLatency withLatency = getWithLatency(command);
 
-            if (!rsm.decode(buffer, command, command.getOutput())) {
-                return;
+            try {
+                if (!rsm.decode(buffer, command, command.getOutput())) {
+                    return;
+                }
+            } catch (Exception e) {
+                rsm.reset();
```

Comment:
This change does not close the Redis connection. `catch` should only complete the command with the causing exception and close the connection (`channel.close()`). `RedisStateMachine` is reset in `channelInactive`. It would also make sense to clear `buffer` in `channelInactive`.